arch/sim/src/sim/sim_initialize.c

```diff
@@ -162,6 +162,8 @@ static int sim_loop_task(int argc, char **argv)
 {
   while (1)
     {
+      irqstate_t flags = up_irq_save();
```

Comment:
should we use enter_critical_section and sched_lock here to simulate the irq context